src/note.rs

```diff
@@ -13,7 +13,9 @@ use dusk_jubjub::{
     dhke, JubJubAffine, JubJubExtended, JubJubScalar, GENERATOR_EXTENDED,
     GENERATOR_NUMS_EXTENDED,
 };
-use dusk_poseidon::cipher::PoseidonCipher;
+
+use crate::cipher::{decrypt, encrypt, EXTRA_DATA_SIZE};
+
 use dusk_poseidon::sponge::hash;
```

Comment:
not directly linked to this PR but are we still using the poseidon hash here?
Cause if not I think we can remove the `dusk-poseidon` dependency entirely!

Reply:
still needed to compute the note hash!